fitbenchmarking/controllers/aoclda_controller.py

```diff
@@ -0,0 +1,327 @@
+            )
+
+        # If parameter ranges have been set in problem, then set up bounds
+        # option. For AOCLDA, this must be a 2 tuple array like object,
+        # the first tuple containing the lower bounds for each parameter
+        # and the second containing all upper bounds.
+        if self.value_ranges is not None:
+            self._lb, self._ub = zip(*self.value_ranges)
+
+        self._handle = nlls(
+            n_coef=n_coef,
+            n_res=n_res,
+            weights=None,
+            lower_bounds=self._lb,
+            upper_bounds=self._ub,
+            order="c",
+            model=model,
+            method=method,
+            glob_strategy=glob_strategy,
+            verbose=self.VERBOSE,
+        )
+
+        # Set initial iterate
+        self._x = np.array(
+            self.initial_params
+        )  # x has to be np.array and is overwritten!
+
+    def fit(self):
+        """
+        Run problem with AOCLDA.
+        Note: only this method is timed
+        """
+
+        try:
+            self._handle.fit(
+                x=self._x,
+                fun=self.eval_r,
+                jac=self.aoclda_jac,
+                hes=self.aoclda_hes,
+                hep=None,
+                data=self,
+                maxit=self.maxit,
+                xtol=self.xtol,
+                # ftol=self.ftol,
+                # abs_ftol=self.abs_ftol,
+                # gtol=self.gtol,
+                # abs_gtol=self.abs_gtol,
+                # fd_step = self.fd_step,
+                # fd_ttol = self.fd_ttol
+            )
+
+        except RuntimeWarning:  # FIXME this is not caught
+            #    2: Software run but didn't converge to solution
+            #    6: Solver has exceeded maximum allowed runtime
+            self._status = 2
+
+        except Exception as e:
+            if self.VERBOSE > 0:
+                print(e)
+                print(traceback.format_exc())
+            self._status = 3  # Software raised an exception
+
+        else:
+            # 0: Successfully converged
+            # Taken care by framework
+            # 1: Software reported maximum number of iterations exceeded
+            # 5: Solution doesn't respect parameter bounds -> taken care by framework
+            self._status = 0
+
+    def cleanup(self):
+        """
+        Convert the result to a numpy array and populate the variables results
+        will be read from.
+        """
+
+        if (
+            self._status == 0
+            or self._status == 1
+            or self._status == 2
+            or self._status == 5
+        ):
+            self.iteration_count = self._handle.n_iter
+            self.func_evals = self._handle.n_eval["f"]  # "fd_f" also available
+
+        # Final Params: The final values for the params from the minimizer
+        self.final_params = self._x.tolist()
+
+        # Flag: error handling flag
+        self.flag = self._status
```

Comment:
Here, we just need to make sure that the we're setting either flag to either 0, 1, or 2. At the moment, I don't think `self._status` gets set to 1 if max iterations are exceeded? For an example of how this is handled in other controllers, see [here](https://github.com/fitbenchmarking/fitbenchmarking/blob/f9b05d41dfe5c104288528ed64ce61080e8da7ce/fitbenchmarking/controllers/scipy_controller.py#L147)